app/views/admin/pages/_form.html.erb

```diff
@@ -38,9 +38,43 @@
     </div>
     <div class="form-group">
       <%= form.label :is_top_level_path %>
-      <%= form.check_box :is_top_level_path %>
-      <p>(Determines if it is accessible by <code>/page-slug</code> vs <code>/page/page-slug</code>) Be careful! ⚠️</p>
+      <%= form.check_box :is_top_level_path, class: "crayons-checkbox" %>
+      <p>Determines if it is accessible by <code>/page-slug</code> vs <code>/page/page-slug</code> Be careful! ⚠️</p>
     </div>
+    <% if @page.errors.count > 0 %>
+      <div id="add-landing-page-link-modal-root">
+        <% @page.errors %>
+      </div>
+    <% end %>
+      <span data-controller="modal" data-modal-root-selector-value="#add-landing-page-link-modal-root" data-modal-content-selector-value="#add-landing-page-link-modal"
+            data-modal-title-value="There's another locked screen..." data-modal-size-value="default">
```

Comment:
I'm a bit confused by the modal code but admittedly I had to struggle a bit on how to make it work for the "pin posts to feed" as well. 

My suggestion would be to add a partial similar to https://github.com/forem/forem/blob/main/app/views/admin/articles/_pinned_article_modal.html.erb and render it like this:

```erb
<% if @landing_page.present? %>
   <%= render partial: "landing_page_modal", locals: { page: @landing_page } %>
<% end %>
```

something like this :D

Reply:
There are a lot of moving pieces that make this modal work and I'm finding myself struggling to get the modal to work properly with so much of this pulled into a partial. 😕